### ISSUE TITLE
Visual Studio 17rc C++ issue

### ISSUE THREAD

**Issue by effolkronium** [Morwenn/static_math#12]
please check the VS 17. There is a some problems (with **sqrt** for example) when building this project. But there are already available C++11|14 constexpr futures. https://www.visualstudio.com/vs/visual-studio-2017-rc/ some functions work. some, like sqrt -- not = (

**Comment by Morwenn** [comment]
I'll try to have a look later, but this project was never meant to work with MSVC. I guess that the most recent versions do have a more decent `constexpr` support than the previous ones, and maybe it wuld be possible to make this library work with them.

That said, unless somebody wants to take the job, I'd have to learn everything about MSVC (never used it), and to set up an AppVeyor for continuous integration (never done that either).

**Comment by Morwenn** [comment]
So: I installed the MSVC build tools, the 2015 ones though since the MSVC 2017 build tools aren't available. I hope they'll be available soon because the 2015 compiler support seems way beyond broken with the library, at least for the following things:
* Variadic templates (pack expansions sometimes don't work)
* `constexpr`
* Lack of alternate keywords of `<ciso646>` by default (I know that's by design, and this can be easily fixed by including the header)
* More than trivial expressions in non-type template parameters
* I'm not sure whether I've got expression SFINAE problems or not

Long story short, I won't make an effort for MSVC 2015. I'll see how things have improved with MSVC 2017 when I've got time (it seems that the build tools are not available alone, so installing the whole thing might take some time). I believe that the `constexpr` support is better, and the variadics may have been partially fixed.

**Comment by Morwenn** [comment]
I have started a branch to make the library work with the latest VS 2017. Things mostly work, but I had to disable the support for integral constants (`smath::constant`) because there were still a few unresolved issues with them.

I need to fix some warnings too, but it seems that most of the functions just work.

**Comment by Morwenn** [comment]
I merged the branch with integral constants disabled. I don't work much on this library, so at least we've now got some basic support for the latest Visual Sutio versions.
